spec/lib/msf/core/opt_array_spec.rb

```diff
@@ -0,0 +1,263 @@
+# -*- coding:binary -*-
+
+require 'spec_helper'
+
+RSpec.describe Msf::OptArray do
+  let(:required_opt) { described_class.new('TestArray', [true, 'A test array', 'foo,bar']) }
+  let(:not_required_opt) { described_class.new('TestArray', [false, 'A test array', 'foo,bar']) }
+  let(:accepted_opt) { described_class.new('TestArray', [true, 'Extensions', 'stdapi,priv'], accepted: %w[stdapi priv incognito]) }
+  let(:case_sensitive_opt) { described_class.new('TestArray', [true, 'Case sensitive', 'Foo,Bar'], accepted: %w[Foo Bar foo bar]) }
+  let(:pipe_separator_opt) { described_class.new('TestArray', [true, 'Pipe separated', 'foo|bar'], separator: '|') }
+  let(:no_unique_opt) { described_class.new('TestArray', [true, 'Allow duplicates', 'foo,bar'], unique: false) }
+
+  it_behaves_like 'an option', [], [], 'array'
+
+  describe '#type' do
+    it 'returns array' do
+      expect(required_opt.type).to eq('array')
+    end
+  end
+
+  context 'initialization' do
+    it 'accepts accepted parameter' do
+      opt = described_class.new('Test', [true, 'desc', 'val'], accepted: %w[val1 val2])
+      expect(opt.accepted).to eq(%w[val1 val2])
+    end
+
+    it 'accepts separator parameter' do
+      opt = described_class.new('Test', [true, 'desc', 'val'], separator: '|')
+      expect(opt.normalize('a|b|c')).to eq(%w[a b c])
+    end
+
+    it 'accepts strip_whitespace parameter' do
+      opt = described_class.new('Test', [true, 'desc', 'val'], strip_whitespace: false)
+      expect(opt.normalize(' a , b ')).to include(' a ')
+    end
+
+    it 'accepts unique parameter' do
+      opt = described_class.new('Test', [true, 'desc', 'val'], unique: false)
+      expect(opt.normalize('a,a,b')).to eq(%w[a a b])
+    end
+  end
+
+  context 'validation when required' do
+    it 'returns false for nil value' do
+      expect(required_opt.valid?(nil)).to eq(false)
+    end
+
+    it 'returns false for empty string' do
+      expect(required_opt.valid?('')).to eq(false)
+    end
```

Comment:
This test is also failing. The `valid?` method calls `empty_required_value?` from `opt_base.rb` which is slightly unintuitive as it checks if the value is `nil` and required but not if it's an empty string and required. 

https://github.com/rapid7/metasploit-framework/blob/a99a74c6f65fe677e387ef87e88fd5667d568705/lib/msf/core/opt_base.rb#L134-L136

We should likely update the `valid?` method to also check for empty strings.